src/__tests__/SearchData.test.js

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import TableContainer from '../components/pages/Table/TableContainer';
+import { legacy_createStore as createStore } from 'redux';
+import { Provider } from 'react-redux';
+
+import reducers from '../state/reducers';
+
+//anywhere you test antd you must add this window.matchMedia
+window.matchMedia =
+  window.matchMedia ||
+  function () {
+    return {
+      matches: false,
+      addListener: function () {},
+      removeListener: function () {},
+    };
+  };
+
+//creates store for testing redux can be moved to separate component if needed and imported in
+function renderWithRedux(
+  ui,
+  { initialState, store = createStore(reducers, initialState) } = {}
+) {
+  return {
+    ...render(<Provider store={store}>{ui}</Provider>),
+    store,
+  };
+}
+
+beforeEach(() => {
+  renderWithRedux(<TableContainer />);
+});
+
+describe('searching through data shows correct filtered data, advanced search', () => {
+  test('Should render App without crashing', async () => {
+    renderWithRedux(<TableContainer />);
+  });
+
+  test('shows original data of 25 entries', () => {
+    const tableRows = document.querySelectorAll('.ant-table-row');
+    expect(tableRows.length).toBe(25);
+  });
+
+  test('shows correct number of results when filtering zch and chad and Deny,Referral', async () => {
```

Comment:
Use of async/await hee rather than .then() was a wise choice, as evidenced by the clarity of this code.